packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -360,10 +360,17 @@ export class Pipeline extends PipelineBase {
     });
   }
 
+  private requireRegion(error?: string) {
+    const region = Stack.of(this).region;
+    if (Token.unresolved(region)) {
+      throw new Error(error || 'region is required');
```

Comment:
This default error message is not good enough. There is *always* a region, only it isn't always known at synthesis time...